### DISCUSSION TITLE
Block PR using Rulesets Metadata restrictions

### DISCUSSION THREAD

**Discussion by dimich99** [community/community#51987] [Pull Requests]
### Select Topic Area

Question

### Body

Hi All. 
I want to block PR using Releset Metadata restriction: Commit message must start with a matching pattern
[e2e]. 
PR is blocked until Bamboo executes e2e tests and if successful it commits back to branch with message "[e2e] passed".
![image](https://user-images.githubusercontent.com/40930080/229892049-93f67c24-d353-43b9-93c6-57ca9e80445d.png)

![image](https://user-images.githubusercontent.com/40930080/229892104-1d4dbd90-90bd-4552-b6b6-7ed3912429eb.png)
The problem: this works as long as at least one message does not start with a word other than [e2e].
In my opinion it should check only last commit message
![image](https://user-images.githubusercontent.com/40930080/229894122-3777d25a-522a-4074-add3-55a0673cc3a0.png)

![image](https://user-images.githubusercontent.com/40930080/229894305-d2a1fede-1466-4147-8f3c-1732ceb3b9ab.png)

![image](https://user-images.githubusercontent.com/40930080/229894733-8ad65917-6b1a-450e-a21f-f5f0494357f6.png)

Is this expected behavior? Am I doing something wrong?

**Comment by patrick-knight** [comment]
This is the expected behavior as the rule evaluation is done on all new inbound commits regardless of the merge method. We plan to move the merge method as a rule in the engine in the future which will allow more more flexibility in how metadata rules interact with merge types.

**Comment by sethuraman0** [comment]
I am also getting the same. as long as at least one message does not start with a word other than rules (our mentioned ruleset) it doesn't works. Any leads would be helpful.

**Reply by patrick-knight** [reply]
See some follow up discussion here: https://github.com/orgs/community/discussions/52652#discussioncomment-5639858

But this is known behavior in rules.
